proxy.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const processFn = (fn, opts) => function () {
+	const that = this;
+	const P = opts.promiseModule;
+	const args = new Array(arguments.length);
+
+	for (let i = 0; i < arguments.length; i++) {
+		args[i] = arguments[i];
+	}
```

Comment:
Alright, let's just hold off on changing that until Node.js 8 which should fix the perf deoptimization for that.